frontend/src/components/game/Editor.tsx

```diff
@@ -8,9 +8,11 @@ import { DefaultCodeType } from '../../api/Problem';
 type EditorProps = {
   onLanguageChange: ((language: Language) => void) | null,
   onCodeChange: ((code: string) => void) | null,
-  codeMap: DefaultCodeType | null,
+  getCurrentLanguage: (() => Language) | null,
+  defaultCodeMap: DefaultCodeType[] | null,
   defaultLanguage: Language,
   defaultCode: string | null,
+  currentProblem: number,
   liveCode: string | null,
```

Comment:
Could we leave a multi-line comment above this editor props carefully detailing what each prop does? On initial glance it can be hard to remember what each of these things does

Also, why is a getCurrentLanguage,  defaultLanguage, and currentLanguage all needed? Could we just pass in the current language as a prop and use that to replace the state variable, or does that cause any bugs?